src/cursor/sys/unix.rs

```diff
@@ -29,15 +33,43 @@ fn read_position() -> Result<(u16, u16)> {
 fn read_position_raw() -> Result<(u16, u16)> {
     // Use `ESC [ 6 n` to and retrieve the cursor position.
     let mut stdout = io::stdout();
-
     stdout.write_all(b"\x1B[6n")?;
     stdout.flush()?;
 
-    let mut reader = TerminalInput::new().read_sync();
-
     loop {
-        if let Some(InputEvent::CursorPosition(x, y)) = reader.next() {
-            return Ok((x, y));
+        let mut temp_buffer = VecDeque::new();
+
+        match poll_internal(Some(Duration::from_millis(2000))) {
+            Ok(true) => {
+                match read_internal() {
+                    Ok(InternalEvent::CursorPosition(x, y)) => {
+                        if !temp_buffer.is_empty() {
+                            while let Some(event) = temp_buffer.pop_front() {
```

Comment:
This `enqueue_internal` is something I don't like much, will think about it.

But there's one thing - this `while` loop & `enqueue_internal` should be called for every return path - `Ok(false)`, `Err(e)`, ... - in other words - in every place where the `read_position_raw` exits.